test/game_connection/connector/test_msr_remote_connector.py

```diff
@@ -0,0 +1,167 @@
+    connector.new_inventory_received("{}")
+    assert connector.last_inventory == Inventory.empty()
+    inventory_updated.assert_not_called()
+
+    assert connector.inventory_index is None
+    connector.new_inventory_received('{"index": 69, "inventory": [0,1,0,1]}')
+    assert connector.inventory_index == 69
+
+    # check ice beam
+    plasma_beam = connector.game.resource_database.get_item_by_name("Ice Beam")
+    assert connector.last_inventory[plasma_beam].capacity == 1
+
+    # check wave beam
+    plasma_beam = connector.game.resource_database.get_item_by_name("Wave Beam")
+    assert connector.last_inventory[plasma_beam].capacity == 0
+
+    # check spazer beam
+    spazer_beam = connector.game.resource_database.get_item_by_name("Spazer Beam")
+    assert connector.last_inventory[spazer_beam].capacity == 1
+
+    inventory_updated.assert_called_once()
+
+
+async def test_new_received_pickups_received(connector: MSRRemoteConnector):
+    connector.receive_remote_pickups = AsyncMock()
+    connector.in_cooldown = True
+
+    await connector.new_received_pickups_received("6")
+    assert connector.received_pickups == 6
+    assert connector.in_cooldown is False
+
+
+async def test_set_remote_pickups(connector: MSRRemoteConnector, msr_ice_beam_pickup):
+    connector.receive_remote_pickups = AsyncMock()
+    pickup_entry_with_owner = (("Dummy 1", msr_ice_beam_pickup), ("Dummy 2", msr_ice_beam_pickup))
+    await connector.set_remote_pickups(pickup_entry_with_owner)
+    assert connector.remote_pickups == pickup_entry_with_owner
+
+
+async def test_receive_remote_pickups(connector: MSRRemoteConnector, msr_ice_beam_pickup):
+    connector.in_cooldown = False
+    pickup_entry_with_owner = (("Dummy 1", msr_ice_beam_pickup), ("Dummy 2", msr_ice_beam_pickup))
+    connector.remote_pickups = pickup_entry_with_owner
+    connector.executor.run_lua_code = AsyncMock()
+
+    connector.received_pickups = None
+    connector.inventory_index = None
+    await connector.receive_remote_pickups()
+    assert connector.in_cooldown is False
+
+    connector.received_pickups = 1
+    connector.inventory_index = None
+    await connector.receive_remote_pickups()
+    assert connector.in_cooldown is False
+
+    connector.received_pickups = None
+    connector.inventory_index = 1
+    await connector.receive_remote_pickups()
+    assert connector.in_cooldown is False
+
+    # FIXME: No module named 'open_samus_returns_rando.multiworld_integration'
+    # try: AttributeError: module 'open_samus_returns_rando' has no attribute 'multiworld_integration'
+    # mocker.patch("open_samus_returns_rando.multiworld_integration.get_lua_for_item", return_value="Hallo")
+    # try:   AttributeError: <module 'open_samus_returns_rando' from 'BLABLA....open_samus_returns_rando\\__init__.py'>
+    #           does not have the attribute 'multiworld_integration'
+    # mocker.patch("open_samus_returns_rando.multiworld_integration", return_value="Hallo")
```

Comment:
Could anyone tell me how I could patch this in? I struggled and didn't figured out (even with searching for it) how to to add `multiworld_integration` into the package when it - atm - doesn't even exist.
I need to mock it for the code test and it will only be part of a future OSRR version

Reply:
I think you should just wait until it's in OSRR